src/core/datatypes/phylogenetics/character/DiscretizedContinuousState.h

```diff
@@ -0,0 +1,75 @@
+        DiscretizedContinuousState(const std::string &s, std::vector<double> pts, int m, double d);     //!< Constructor with an observation
+        DiscretizedContinuousState(int s, std::vector<double> pts, int m, double d);                    //!< Constructor with an observation
+                
+        DiscretizedContinuousState*            clone(void) const;                                  //!< Get a copy of this object
+        
+        // Discrete character observation functions
+        void                            addState(const std::string &symbol);                //!< Add a character state to the set of character states
+        RbBitSet                        getState(void) const;                               //!< Get the state (as the bitset)
+        void                            setToFirstState(void);                              //!< Set this character state to the first (lowest) possible state
+        void                            setState(const std::string &symbol);                //!< Compute the internal state value for this character.
+        void                            setStateByIndex(size_t index);                      //!< Set the discrete observation
+
+        void                            addState(int s);                                    //!< Add the state with the given index.
+        void                            addStateDescriptions(const std::vector<std::string>& d);
+        std::string                     getDataType(void) const;                            //!< Get the datatype as a common string.
+        double                          getDeltaX(void) const;
+        std::vector<double>             getPoints(void) const;
+        std::string                     getStateDescription(void) const;
+        std::vector<std::string>        getStateDescriptions(void) const;
+        std::string                     getStateLabels(void) const;                         //!< Get valid state labels
+        std::string                     getStringValue(void) const;                         //!< Get a representation of the character as a string
+        bool                            isGapState(void) const;                             //!< Get whether this is a gapped character state
+        bool                            isMissingState(void) const;                         //!< Get whether this is a missing character state
+        void                            setGapState(bool tf);                               //!< set whether this is a gapped character
+        void                            setMissingState(bool tf);                           //!< set whether this is a missing character
+        void                            setWeights(std::vector<double> w);                  //!< set the weight for each character
+        
+    private:
+        
+        bool                            is_gap;
```

Comment:
Maybe initialize this to false here - `bool is_gap = false`.

Reply:
Fantastic, done for all member variables that need defaults.